lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -336,18 +342,17 @@ def change_table(model, current_table_name)
           new_table_name = model.table_name
 
           db_columns = model.connection.columns(current_table_name).index_by(&:name)
-          key_missing = db_columns[model._defined_primary_key].nil? && model._defined_primary_key.present?
-          if model._defined_primary_key.present?
-            db_columns.delete(model._defined_primary_key)
+          if (pk = model._declared_primary_key.presence)
+            key_was_in_db_columns = db_columns.delete(pk)
```

Comment:
I finally untangled this rat's nest. Turns out it was much simpler than it looked.